crates/payload/basic/src/lib.rs

```diff
@@ -828,6 +813,21 @@ pub trait PayloadBuilder<Pool, Client>: Send + Sync + Clone {
         client: &Client,
         config: PayloadConfig<Self::Attributes>,
     ) -> Result<Self::BuiltPayload, PayloadBuilderError>;
+
+    /// Returns the configured [`CfgEnvWithHandlerCfg`] and [`BlockEnv`] for the targeted payload
+    /// (that has the `parent` as its parent).
+    ///
+    /// The `chain_spec` is used to determine the correct chain id and hardfork for the payload
+    /// based on its timestamp.
+    ///
+    /// Block related settings are derived from the `parent` block and the configured attributes.
+    ///
+    /// NOTE: This is only intended for beacon consensus (after merge).
```

Comment:
I don't have enough context to meaningfully update this comment so please suggest alternatives.